### PR TITLE
refactor for Admin, Hr and User Profile.

### PR DESCRIPTION
**Pull request by DhyanKPatel**

### CODE REVIEW

File: src/adminProfile/adminProfile.controller.ts [hunk 1]
```diff
@@ -31,43 +23,17 @@ export class AdminProfileController {
   @Roles(Role.Admin, Role.HR)
   @UseGuards(JwtGuard, RolesGuard)
   @ApiBearerAuth()
+  @ApiOperation({ summary:'Admin and Hr can add Employee.'})
   @Post('addEmployee')
   addEmployee(@Body() dto: AddEmployeeDto) {
```
Comment (32KAR) [inline src/adminProfile/adminProfile.controller.ts:28]: Give return type of functions also.

File: src/adminProfile/adminProfile.service.ts [hunk 2]
```diff
@@ -1,321 +1,127 @@
...
-    @InjectModel(State) private stateModel: typeof State,
-    @InjectModel(Country) private countryModel: typeof Country,
-    @InjectModel(Leave) private leaveModel: typeof Leave,
+    @InjectModel(AuthUser) private authUserModel: typeof AuthUser,
     @InjectModel(ReportTo) private reportToModel: typeof ReportTo,
     @InjectModel(CasualLeave) private casualLeaveModel: typeof CasualLeave,
     private jwt: JwtService,
   ) {}
 
   async addEmployee(dto: AddEmployeeDto) {
```
Comment (32KAR) [inline src/adminProfile/adminProfile.service.ts:23]: Return give return type of functions.

File: src/services/serviceLayer.ts [hunk 3]
```diff
@@ -56,4 +57,13 @@ export function Filter(
...
\ No newline at end of file
+};
+
+export class StatusRO {
+  @ApiProperty({ example: 200 })
+  statusCode?: number;
+  @ApiProperty({ example: "successfully" })
+  message?: string;
+  data?:any;
+  error: any;
```
Comment (32KAR) [inline src/services/serviceLayer.ts]: Is this mandatory?

File: src/services/dbService.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    collection: any,
+    condition: any,
+    selectAttribute: any[],
+    include?: any[],
+  ) {
+    try {
+      return collection.findOne({
+        attributes: selectAttribute,
+        where: condition,
+        include: include,
```
Comment (32KAR) [inline src/services/dbService.ts:15]: When key and value are same so give only variable name.

File: src/services/dbService.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import { Injectable } from '@nestjs/common';
+
+@Injectable()
+export class dbService {
+  async findOne(
+    collection: any,
+    condition: any,
+    selectAttribute: any[],
```
Comment (32KAR) [inline src/services/dbService.ts:8]: Give optional chaining for this.

File: src/hrProfile/hrProfile.service.ts [hunk 6]
```diff
@@ -156,184 +165,139 @@ export class HrProfileService {
...
-          userId: id,
-        },
-      })
-      .catch((err) => {
-        error = err;
-      });
+      const addressExists: AddressModel = await this.dbService.findOne(
+        this.addressModel,
+        { userId: id },
+        undefined,
```
Comment (32KAR) [inline src/hrProfile/hrProfile.service.ts:191]: Remove undefine when you put at the end.

File: src/adminProfile/adminProfile.service.ts [hunk 7]
```diff
@@ -1,321 +1,150 @@
...
-        undefined,
-      );
-    }
   }
 
-  async viewProfile(adminId: number) {
-    let error = null;
+  async login(dto: AdminLoginDto): Promise<StatusRO> {
+    try {
+      const { email } = dto;
```
Comment (32KAR) [inline src/adminProfile/adminProfile.service.ts:90]: Give type for email.

File: src/adminProfile/adminProfile.service.ts [hunk 8]
```diff
@@ -1,321 +1,150 @@
...
-          errorMessage: error.original.sqlMessage,
-          field: error.fields,
-        },
       );
-    }
 
-    if (findAdminDetails && Object.keys(findAdminDetails).length > 0) {
-      delete findAdminDetails.dataValues.password;
+      if (employeeHrAdminLogin) {
+        const comparison = await bcrypt.compare(
```
Comment (32KAR) [inline src/adminProfile/adminProfile.service.ts:99]: Give type here also.

File: src/adminProfile/adminProfile.service.ts [hunk 9]
```diff
@@ -1,321 +1,150 @@
...
-      const findCountry: any = await this.countryModel
-        .findOne({
-          where: {
-            id: findAdminDetails.countryId,
-          },
-        })
-        .catch((err) => {
-          error = err;
-        });
+        const id = employeeHrAdminLogin.id;
```
Comment (32KAR) [inline src/adminProfile/adminProfile.service.ts:104]: Give type here also.

File: src/adminProfile/adminProfile.service.ts [hunk 10]
```diff
@@ -1,321 +1,150 @@
...
+        const role = employeeHrAdminLogin.role;
```
Comment (32KAR) [inline src/adminProfile/adminProfile.service.ts:105]: Give type here also.

File: src/adminProfile/adminProfile.service.ts [hunk 11]
```diff
@@ -1,321 +1,150 @@
...
 
-      if (error) {
+        if (comparison) {
+          const token = await this.jwt.signAsync({
```
Comment (32KAR) [inline src/adminProfile/adminProfile.service.ts:108]: Give type here also.

File: src/adminProfile/adminProfile.service.ts [hunk 12]
```diff
@@ -1,321 +1,150 @@
...
-            id: adminId,
-          },
-        },
-      )
-      .catch((err) => {
-        error = err;
-      });
-
-    if (error) {
+    } catch (error) {
```
Comment (32KAR) [inline src/adminProfile/adminProfile.service.ts:137]: Give type of error variable.

File: src/hrProfile/hrProfile.service.ts [hunk 13]
```diff
@@ -35,117 +44,123 @@ export class HrProfileService {
...
-          assigneeId: id
-        },
-        include: [{
-          model: this.userModel,
-          attributes: ['firstName','lastName']
-        }]
-      }).catch((err) => {
-        error = err;
-      });
+      findDetails.dataValues.reportTo = findReportToDetails?.authUser
```
Comment (32KAR) [inline src/hrProfile/hrProfile.service.ts:146]: Give type here also to "findDetails.dataValues.reportTo".

File: src/hrProfile/hrProfile.service.ts [hunk 14]
```diff
@@ -35,117 +44,123 @@ export class HrProfileService {
...
+      } else {
+        Logger.error(Messages.NOT_FOUND);
+        return HandleResponse(
+          HttpStatus.NOT_FOUND,
+          Messages.NOT_FOUND,
+          undefined,
+          undefined,
+        );
+      }
+    } catch (error) {
```
Comment (32KAR) [inline src/hrProfile/hrProfile.service.ts:162]: Give type here also.

File: src/hrProfile/hrProfile.service.ts [hunk 15]
```diff
@@ -156,184 +171,141 @@ export class HrProfileService {
...
+          });
+
+      if (dto.reportTo) {
+        await this.dbService.update(
+          this.reportToModel,
+          { assignerId: dto.reportTo },
+          { assigneeId: dto.assigneeId },
+        );
+      }
+      if (
```
Comment (32KAR) [inline src/hrProfile/hrProfile.service.ts:240]: Give space above.

File: src/hrProfile/hrProfile.service.ts [hunk 16]
```diff
@@ -156,184 +171,141 @@ export class HrProfileService {
...
-        (updateAddressDetails[0] === 1 && updateEmergencyDetails[0] === 1))|| addedReportTo[0] === 1
-    ) {
-      return HandleResponse(
-        HttpStatus.OK,
-        `Your profile ${Messages.UPDATE_SUCCESS}`,
-        undefined,
-        undefined,
-      );
-    } else {
+    } catch (error) {
```
Comment (32KAR) [inline src/hrProfile/hrProfile.service.ts:265]: Give type here also.

File: src/hrProfile/hrProfile.service.ts [hunk 17]
```diff
@@ -156,184 +171,141 @@ export class HrProfileService {
...
+      } else {
+        Logger.error(Messages.ID_NOT_FOUND);
+        return HandleResponse(
+          HttpStatus.NOT_FOUND,
+          Messages.ID_NOT_FOUND,
+          undefined,
+          undefined,
+        );
+      }
+    } catch (error) {
```
Comment (32KAR) [inline src/hrProfile/hrProfile.service.ts:307]: Give type here also.

File: src/hrProfile/hrProfile.service.ts (hunk 13 above)
Comment (DhyanKPatel) [reply src/hrProfile/hrProfile.service.ts:146]: only refers to a type, but is being used as a value here.

**Pull request merged**